index.bs

```diff
@@ -161,19 +162,16 @@ To <dfn abstract-op local-lt="parse header">obtain a response's embedder policy<
 
 3.  If |header| is not `null`:
 
-    1.  Set |header| to the result of [=isomorphic decoding=] |header|.
+    1.  Let |parsed policy| and |parsed parameters| be the result of executing
+        [$Parsing Header Fields into Structured Headers$] with <var ignore>input_string</var> set to
+        |header|, and <var ignore>header_type</var> set to "`item`".
 
-    2.  Let |parsed policy| be the result of executing the [$Structured Header parsing algorithm$]
-        with <var ignore>input_string</var> set to |header|, and <var ignore>header_type</var> set
-        to "`item`".
+        If |parsed policy| is not a token, fail parsing.
 
-        If parsing fails, set |parsed policy| to "`none`".
+        If |parsed parameters| is not empty, fail parsing.
 
-        ISSUE(httpwg/http-extensions#662): The ASCII requirements of Structured Headers are
-        somewhat unclear to me. Do we need to check whether |header| is an [=ASCII string=] before
-        passing it into the parsing algorithm?
-
-    3.  If |parsed policy| is "`require-corp`", set |policy| to "`require-corp`".
+    2.  If parsing did not fail and |parsed policy| is "`require-corp`", then set |policy| to
```

Comment:
This is fine for the moment, but I don't really like the implicitness of "parsing failed". I'd prefer for the parsing algorithm to return something that we interpret as failure. Like `null`? (Obviously, this isn't the first place we're using this pattern, URL parsing comes to mind as well, but I don't like it there either. :) )

Reply:
@annevk @mnot

Reply:
I don't really mind how Structured Headers words it, but I agree that whatever we add in Fetch will have to do something like that. Bridging with the IETF world is always going to look a little clunky I suppose, though this is miles ahead of the pack.

Reply:
I see now that URL parser does "return failure", which is nicer than I remember. That explicit model seems reasonable to me. It might be nice to define that pattern via Infra?

Reply:
@mikewest could you open an issue on SH regarding this?